services/api/src/resources/problem/resolvers.ts

```diff
@@ -9,17 +9,29 @@ const logger = require('../../logger');
 export const getAllProblems: ResolverFn = async (
   root,
   args,
-  { sqlClient, hasPermission }
+  {
+    sqlClient,
+    hasPermission,
+    models,
+    keycloakGrant
+  },
 ) => {
   let rows = [];
 
   try {
-    if (!R.isEmpty(args)) {
-      rows = await problemHelpers(sqlClient).getAllProblems(args.source, args.environment, args.envType, args.severity);
-    }
-    else {
-      rows = await query(sqlClient, Sql.selectAllProblems({source: [], environmentId: 0, environmentType: [], severity: []}));
-    }
+    const userProjectIds = await models.UserModel.getAllProjectsIdsForUser({
+      id: keycloakGrant.access_token.content.sub,
+    });
```

Comment:
This isn't a full proof way to return all the projects the current user has access to. Changing this now might not be a blocker but might need to be addressed at some point down the line.

The first issue is the reliance on `keycloakGrant`. This is only passed to the resolver in the case where the user querying the API is logging in via keycloak. We still have a legacy JWT token system that doesn't use keycloak at all, and in that case this will probably throw an error. In the majority if cases, only other system services use that token so I imagine you might not see issues right now with everything being UI based.

The second issue is usage of the API by lagoon admins. Those user accounts aren't added to all the groups, they just get an admin role. So in this case the method will return 0 projects and admins won't be able to see any data. The way we get around this in other resolvers is to create a new scope `viewAll` which we grant to only admins. If a user can `viewAll` projects then just do `select * from` otherwise only load the data the user has access to. For this issue, it looks like you might need to check twice, once for `project viewAll` and then add a new `viewAll` scope for `problem` and check for that also? The best example is probably the [`getAllProjects`](https://github.com/amazeeio/lagoon/blob/main/services/api/src/resources/project/resolvers.ts#L26) resolver.

Reply:
@rocketeerbkw thanks for the detailed response 👍 
@bomoko something for us to discuss:
1. question is, do we address the non-keycloak/legacy auth issue now - or come back to it?
2. we did originally implement a `viewAll` permission for problems, but went against it on the basis that all problems are directly linked to the same project check anyway. Would only a `project viewAll` permission scope be required?